planner/core/plan.go

```diff
@@ -283,13 +294,61 @@ func (p *basePhysicalPlan) ExtractCorrelatedCols() []*expression.CorrelatedColum
 	return nil
 }
 
+func (p *baseLogicalPlan) GetBakTimeStamp() time.Time {
+	return time.Now()
+}
+
+func (p *baseLogicalPlan) rollBackTaskMap(TS time.Time) {
```

Comment:
From my understanding, the purpose of this function is to reset the `taskMap` cache in order to re-enumerate the child tasks in `rebuildChildTasks`, then we may have a simpler approach to achieve this: add a bool flag parameter for `findBestTask` to indicate whether we check `taskMap` cache?

Reply:
The reason I designed that is exactly "not to introduce more parameters to `findBestTask` "

Reply:
I think an additional parameter can decrease the code complexity and increase the readability.